src/dmstor.cpp

```diff
@@ -27,38 +27,53 @@ dmstor(const char *is, char **rs) {
 
 	double
 dmstor_ctx(PJ_CONTEXT *ctx, const char *is, char **rs) {
-	int n, nl;
+	int n, nl, adv;
 	char *s, work[MAX_WORK];
-        const char* p;
+	const char* p;
 	double v, tv;
 
 	if (rs)
 		*rs = (char *)is;
-	/* copy sting into work space */
+	/* copy string into work space */
 	while (isspace(*is)) ++is;
 	n = MAX_WORK;
 	s = work;
 	p = (char *)is;
-	while (isgraph(*p) && --n)
+
+	/*
+	 * Copy characters into work until we hit a non-printable character or run
+	 * out of space in the buffer.  Make a special exception for the bytes 0xc2
+	 * and 0xb0, because they comprise Degree Sign U+00B0 in UTF-8.
+	 *
+	 * It is possible that a really odd input (like lots of leading zeros)
+	 * could be truncated in copying into work.  But ...
+	 */
+	while ((isgraph(*p) || *p == (char) 0xc2 || *p == (char) 0xb0) && --n)
 		*s++ = *p++;
 	*s = '\0';
-	/* it is possible that a really odd input (like lots of leading
-		zeros) could be truncated in copying into work.  But ... */
 	int sign = *(s = work);
 	if (sign == '+' || sign == '-') s++;
 	else sign = '+';
 	v = 0.;
-	for (nl = 0 ; nl < 3 ; nl = n + 1 ) {
+	for (nl = 0 ; nl < 3 ; nl = n + 1) {
 		if (!(isdigit(*s) || *s == '.')) break;
 		if ((tv = proj_strtod(s, &s)) == HUGE_VAL)
 			return tv;
+		adv = 1;
 		switch (*s) {
 		case 'D': case 'd':
 			n = 0; break;
 		case '\'':
 			n = 1; break;
 		case '"':
 			n = 2; break;
+		/* degree symbol ("\xc2\xb0" in UTF-8) */
+		case (char) 0xc2:
+			if (s[1] == (char) 0xb0) {
+				n = 0;
+				adv = 2;
+				break;
+			}
```

Comment:
there is missing code in the case s[1] is not 0xb0. I guess this should be what the default: case does

Reply:
HI @rouault, if s[1] is not 0xb0, then the code will not 'break', and so it should run through the other branches of the switch statement, and won't find any other matches, so it will flow through to the 'default' case.

At least, that was my intention.  Why do you believe it won't work?

Reply:
>  and so it should run through the other branches of the switch statement, and won't find any other matches, so it will flow through to the 'default' case.

no, it will go on on the ``case 'r': case 'R':``. That's how switch / case works. The pedeantic compiler warnings we turn on in the continuous integration complain about that. You'll likely have to add the code of the ``default`` in the else branch

Reply:
```
../../src/dmstor.cpp: In function 'double dmstor_ctx(PJ_CONTEXT*, const char*, char**)':
../../src/dmstor.cpp:72:4: error: this statement may fall through [-Werror=implicit-fallthrough=]
    if (s[1] == (char) 0xb0) {
    ^~
../../src/dmstor.cpp:77:3: note: here
   case 'r': case 'R':
   ^~~~
```

Reply:
OK, thanks for picking this up.  Looking at it again with your comment in mind, I actually feel like this block might work better as a series of if .. else ifs, rather than a switch.  That way we can test for both bytes together and no need to duplicate the code in the "default" case.